File: app/src/main/java/com/example/fasst_movie/Activity/DetailActivity.java
package com.example.fasst_movie.Activity;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.provider.MediaStore;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.LinearLayout;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CenterCrop;
import com.bumptech.glide.load.resource.bitmap.GranularRoundedCorners;
import com.bumptech.glide.request.RequestOptions;
import com.example.fasst_movie.Adapter.FilmListAdapter;
import com.example.fasst_movie.Domain.Film;
import com.example.fasst_movie.R;
import com.example.fasst_movie.databinding.ActivityDetailBinding;
import com.example.fasst_movie.databinding.ActivityMainBinding;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;
import com.ismaeldivita.chipnavigation.ChipNavigationBar;

import java.util.ArrayList;
import java.util.List;

public class DetailActivity extends AppCompatActivity {
    private ActivityDetailBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityDetailBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        
        setVaiable();

        Window w = getWindow();
        w.setFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS);

        bottomMenu();

    }

    private void bottomMenu() {
        LinearLayout home = findViewById(R.id.home);
        LinearLayout superhero = findViewById(R.id.superhero);
        LinearLayout magic = findViewById(R.id.magic);
        LinearLayout animation = findViewById(R.id.animation);

        home.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(DetailActivity.this, MainActivity.class));
            }
        });
        superhero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(DetailActivity.this, SuperheroActivity.class));
            }
        });
        magic.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(DetailActivity.this, MagicActivity.class));
            }
        });
        animation.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(DetailActivity.this, AnimationActivity.class));
            }
        });

    }

    private void setVaiable() {
        Film item = (Film) getIntent().getSerializableExtra("object");
        RequestOptions requestOptions = new RequestOptions();
        requestOptions = requestOptions.transform(new CenterCrop(), new GranularRoundedCorners(50, 50, 50, 50));

        Glide.with(this)
                .load(item.getPoster())
                .apply(requestOptions)
                .into(binding.picMovie);

        binding.title.setText(item.getTitle());

        binding.duration.setText(item.getDuration());

        binding.summary.setText(item.getSummary());

        binding.backImage.setOnClickListener(v -> finish());

        binding.watchMovieButton.setOnClickListener(view -> {
            Intent intent = new Intent(DetailActivity.this, PlayerActivity.class);
            intent.putExtra("url",item.getVideo());
            startActivity(intent);
        });

    }
}

File: app/src/main/java/com/example/fasstappcoffee/Activity/DetailActivity.kt
package com.example.fasstappcoffee.Activity

import android.content.Context
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import com.example.fasstappcoffee.Model.ItemsModel
import com.example.fasstappcoffee.R
import com.example.fasstappcoffee.databinding.ActivityDetailBinding
import com.example.fasstappcoffee.databinding.ActivityMainBinding

class DetailActivity : BaseActivity() {
    private lateinit var binding: ActivityDetailBinding
    private lateinit var item: ItemsModel
    private lateinit var context: Context

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)
        context = this

        bundle()

    }

    private fun bundle() {
        binding.apply {

            item = intent.getParcelableExtra("object")!!
            titleTxt.text = item.title
            subtitleTxt.text = item.extra
            descriptionTxt.text = item.description
            priceTxt.text = "$" + item.price

            Glide.with(context)
                .load(item.picUrl[0])
                .apply(RequestOptions.bitmapTransform(RoundedCorners(100)))
                .into(binding.img)

            backBtn.setOnClickListener { finish() }

            sizeBtn1.setOnClickListener {
                sizeBtn1.setBackgroundDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.orange_stroke_bg
                    )
                )
                sizeBtn2.setBackgroundDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.dark_grey_bg2
                    )
                )
                sizeBtn3.setBackgroundDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.dark_grey_bg2
                    )
                )
                sizeBtn1.setTextColor(getResources().getColor(R.color.orange))
                sizeBtn2.setTextColor(getResources().getColor(R.color.white))
                sizeBtn3.setTextColor(getResources().getColor(R.color.white))
            }
            sizeBtn2.setOnClickListener {
                sizeBtn1.setBackgroundDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.dark_grey_bg2
                    )
                )
                sizeBtn2.setBackgroundDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.orange_stroke_bg
                    )
                )
                sizeBtn3.setBackgroundDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.dark_grey_bg2
                    )
                )
                sizeBtn1.setTextColor(getResources().getColor(R.color.white))
                sizeBtn2.setTextColor(getResources().getColor(R.color.orange))
                sizeBtn3.setTextColor(getResources().getColor(R.color.white))
            }
            sizeBtn3.setOnClickListener {
                sizeBtn1.setBackgroundDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.dark_grey_bg2
                    )
                )
                sizeBtn2.setBackgroundDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.dark_grey_bg2
                    )
                )
                sizeBtn3.setBackgroundDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.orange_stroke_bg
                    )
                )
                sizeBtn1.setTextColor(getResources().getColor(R.color.white))
                sizeBtn2.setTextColor(getResources().getColor(R.color.white))
                sizeBtn3.setTextColor(getResources().getColor(R.color.orange))
            }
        }
    }
}












